server/Dockerfile

```diff
@@ -47,82 +50,68 @@ FROM "${DOCKER_HUB_PROXY}debian:bullseye-slim" as php-build
-    RUN git clone --depth 1 https://github.com/MISP/cti-python-stix2.git; \
-        cd cti-python-stix2 || exit; python3 setup.py bdist_wheel -d /wheels; \
-        echo "-e ." > requirements.txt; pip3 wheel -r requirements.txt --no-cache-dir -w /wheels/
-
-    # Install the new build tool
-    RUN pip3 install build
-
-    # install PyMISP
-    RUN git clone --depth 1 https://github.com/MISP/PyMISP.git; \
-        cd PyMISP || exit; python3 -m build --wheel --outdir /wheels
-
-    # install pydeep2 (drop-in replacement for pydeep)
-    RUN git clone --depth 1 https://github.com/JakubOnderka/pydeep.git; \
-        cd pydeep || exit; python3 setup.py bdist_wheel -d /wheels
-
-    # Grab other modules we need
-    RUN pip3 wheel --no-cache-dir -w /wheels/ plyara pyzmq redis python-magic lief>=0.13.1
-
-    # Remove extra packages due to incompatible requirements.txt files
-    WORKDIR /wheels
-    RUN find . -name "Sphinx*" | tee /dev/stderr | grep -v "Sphinx-1.5.5" | xargs rm -f
-
+    ARG MISP_TAG
+    ARG MISP_COMMIT
+    # Download MISP using git in the /var/www/ directory. Remove unnecessary items.
+    RUN <<-EOF
+            if [ ! -z "${MISP_COMMIT}" ]; then
+                git clone https://github.com/MISP/MISP.git /var/www/MISP && cd /var/www/MISP && git checkout "${MISP_COMMIT}"
+            else
+                git clone --branch "${MISP_TAG}" --depth 1 https://github.com/MISP/MISP.git /var/www/MISP
+            fi
+
+            cd /var/www/MISP || exit; git submodule update --init --recursive .
+
+            # Until MISP project specifies its required PyMISP version, we grab this from the submodule
+            pymispversion=$(head -n1 PyMISP/pymisp/__init__.py | awk '{print $3}' | tr -d \')
+
+            if [ -n "$pymispversion" ]; then
+                sed -i "s/pymisp$/pymisp==$pymispversion/" requirements.txt
+                cat requirements.txt
+            fi
+            # End of pymisp version fixing hack
+
+            # Remove modules only used during MISP build
+            set -- 'coveralls' 'codecov$' 'requests-mock' 'nose'
+            for mod in "$@"; do
+                sed "/${mod}/d" -i /var/www/MISP/requirements.txt
+            done;
+
+            # Add additional dependencies (container specific)
+            set -- 'redis' 'lief>=0.13.1' 'pydeep2' 'misp-lib-stix2' 'maec' 'mixbox'
+            for mod in "$@"; do
+                grep -qxF ${mod} /var/www/MISP/requirements.txt || echo ${mod} >> /var/www/MISP/requirements.txt
+            done;
+
+            pip3 wheel --no-cache-dir -w /wheels/ -r /var/www/MISP/requirements.txt
+            
+            # Remove files we do not care for
+            rm -r /var/www/MISP/PyMISP
+            find /var/www/MISP/INSTALL/* ! -name 'MYSQL.sql' -type f -exec rm {} +
+            find /var/www/MISP/INSTALL/* ! -name 'MYSQL.sql' -type l -exec rm {} +
+            rm -rf /var/www/MISP/.git
```

Comment:
@oivindoh because of this line (which we might want to keep) the diagnostic page does not display the list of submodules. We should with other MISP devs what is the impact of this

Reply:
I don't really think that's valuable information since we're disabling the update process for our static docker image anyway. Actually, we can also remove the git application itself since MISP is grabbed from prior stages now.